atomicapp/providers/docker.py

```diff
@@ -53,37 +55,49 @@ def init(self):
             logger.info("DRY-RUN: Did not check Docker version compatibility")
         else:
             cmd_check = ["docker", "version"]
-            try:
-                docker_version = subprocess.check_output(cmd_check).split("\n")
-            except Exception as ex:
-                raise ProviderFailedException(ex)
-
-            client = ""
-            server = ""
-            for line in docker_version:
-                if line.startswith("Client API version"):
-                    client = line.split(":")[1]
-                if line.startswith("Server API version"):
-                    server = line.split(":")[1]
-
-            if client > server:
-                msg = ("Docker version in app image (%s) is higher than the one "
-                       "on host (%s). Please update your host." % (client, server))
-                raise ProviderFailedException(msg)
+
+            version = self.shipy(cmd_check)
+
+            compatibility = {
+                '1.21': ('1.6.0', '1.7.0', '1.7.1', '1.7.2'),
+                '1.22': ('1.8.0', '1.8.1'),
+                '1.23': ('1.8.0', '1.8.1')
```

Comment:
That is `{docker api version: docker-py version}`, and IIRC the latest docker API version is 1.23, so every docker version which uses 1.21, 1.22, 1.23 should be fine.